pkg/scheduler/tests/scheduler_utils.go

```diff
@@ -103,6 +103,20 @@ func waitForNodesAllocatedResource(t *testing.T, cache *cache.ClusterInfo, parti
 	}
 }
 
+func waitForNodesAvailableResource(t *testing.T, cache *cache.ClusterInfo, partitionName string, nodeIDs []string, availableMemory resources.Quantity, timeoutMs int) {
+	var totalNodeResource resources.Quantity
+	err := common.WaitFor(10*time.Millisecond, time.Duration(timeoutMs)*time.Millisecond, func() bool {
+		totalNodeResource = 0
+		for _, nodeID := range nodeIDs {
+			totalNodeResource += cache.GetPartition(partitionName).GetNode(nodeID).GetAvailableResource().Resources[resources.MEMORY]
+		}
+		return totalNodeResource == availableMemory
+	})
+	if err != nil {
+		t.Fatalf("Failed to wait for available resource %v and node %v, called from: %s", availableMemory, nodeIDs, caller())
```

Comment:
use `assert.NilError(t, err, "...")

Reply:
yeah, but we might need another cleanup JIRA for this. Given all checks in this file are using a similar way.

Reply:
filed follow up jira: YUNIKORN-53